### DISCUSSION TITLE
When mobile/iPad application?

### DISCUSSION THREAD

**Discussion by wdmason** [brianpetro/obsidian-smart-connections#140] [Ideas]
I recall seeing mobile apps listed as an upcoming feature, but I wonder if we could get more specific information. Over the next week, I'll be transitioning exclusively to the iPad and won't have access to the desktop for several months thereafter. I'm particularly interested in continuing to use the Smart Connections plug-in among other similar tools.

**Comment by brianpetro** [comment]
There's no reason it shouldn't work on mobile. But, it's a computational and data heavy application, so I didn't want to have to make concessions on features early on because of performance on mobile. 

Additionally, since adding the Smart Chat feature, I think that this will be much more useful on mobile compared to when Smart View was the only feature.

The main hold-up is just testing on mobile, and maybe making sure no excess processes are running in the background to prevent draining the battery too quickly.

**Reply by wdmason** [reply]
Sounds like I just need to be patient. You are correct that you will want to tweak and optimize for mobile. Thank you for the feedback.

**Comment by calharding** [comment]
As someone who is presently using an iPad exclusively, I have a vested interest in this as well.

I believe that the Smart Chat feature is coming soon to mobile. May I ask if this implies that other features (particularly Smart Connections) will not be coming to mobile/iPad? Smart Chat seems to be a great feature, but so does Smart Connections especially! :)

If there’s no technical reason why the plugin and its features won’t work on mobile, and your primary concern is, understandably, performance / battery drain on mobile, would it not be a possible option to have the more “resource intensive” features sit behind a toggle, thus giving the user the option?

The iPad Pros, especially, have pretty great hardware (M1/M2 CPU [8 cores], 8/16GB RAM, 10+ hour battery life), so it would be such a shame if the eventual mobile version won’t have at least the option of feature parity with its desktop counterpart.

Keep up the great work!

**Comment by TfTHacker** [comment]
Irony of ironies. I am experimenting with Smart Connections on mobile and in fact used it as an example plugin in this article:

https://tfthacker.medium.com/hacking-obsidian-making-a-desktop-only-plugin-work-on-your-phone-or-tablet-d49ed357b02c

**Reply by calharding** [reply]
Thanks so much for the link! I followed the instructions — though the Smart Chat functionality (mostly) works, the Smart Connections pane just constantly says “Making Smart Connections…XXXX” where “XXXX” is just a counter accumulating without ever stopping and giving me any connections. This is despite the fact that I have <100 notes. This is on an iPad Pro M1.

Have you experienced something similar? If so, have you managed to get around this behaviour?

In any event, thanks again for the input.

**Reply by TfTHacker** [reply]
I have been using only the chat pane. Today I turned on the connections pane, but then noticed something wasn't right with exclusions. So I am waiting to resolve that before trying it.

Also, in theory, if you are using sync, if you did the embeddings file already, that should be used from your desktop. But again, we are hacking :-) around the design, so perhaps it will fail.

**Reply by brianpetro** [reply]
@TfTHacker thanks for doing this! This is exactly what's needed before I can publish the plugin for mobile, testing! 

I'm looking forward to seeing which issues you run into and learning what might be necessary to make Smart Connections available on mobile.

Thanks! 
Brian 🌴

**Reply by dvansoye** [reply]
I can confirm the bug mentioned by @calharding. Chat works great. But connections didn't. I got the same message he did. Making smart connections xxxx. I killed it when it got to 10000 since I didn't remember seeing this message on Windows. 

Edit: it does happen on my laptop. But it occurs very quickly and tops out at 20 or 30 connections. On my phone it continues forever.

And, I only have embeddings for one folder with 11 notes in it. (I excluded all other folders one by one.)

I'm on Android. 

@brianpetro I can create an issue if that would be helpful. I know that mobile is a lower priority. But it might make sense to start a log?

**Comment by houndmaster** [comment]
eta? Would love to have smart connections on iphone

**Reply by brianpetro** [reply]
Currently no ETA.

I have made progress on this.

Smart Connections will now load on mobile, which required replacing some incompatible methods that existed in early versions.

However, a hard-to-debug UI issue has prevented me from moving forward.

I will take another look at it once I get `v2.1` ready for testing (which should be any day now).

🌴

**Reply by houndmaster** [reply]
Sounds excellent! Thank you!

**Reply by Aubermean** [reply]
Any updates now 2.1 is available? I would also like to try this on iPad.

**Reply by brianpetro** [reply]
Not yet, I have a couple of things planned to try, but I'm still trying to get `v2.1` into general release first. 🌴

**Comment by brianpetro** [comment]
@Aubermean @houndmaster 
Mobile is in beta-testing in v2.2 https://twitter.com/wfhbrian/status/1788293763892191528 👀

**Reply by Boombyron** [reply]
@brianpetro I will have you know I am starting to learn Obsidian on iPad starting today only in anticipation of Smart Connections releasing on iPadOS! Oh, and my birthday is in October (wink).
